README.md

```diff
@@ -28,6 +32,123 @@ Project is written in TypeScript and provides type safety out of the box. No Flo
+
+### Adding reactivity to React components
+
+There are three ways to add reactivity to components: `observer(fn)`, `<Observer>{renderFn}</Observer>` and `useObserver(renderFn)`.
+In most cases, there difference doesn't matter that much, but here is an overview of the three different approaches:
+
+```javascript
+import { observable } from "mobx"
+import { Observer, useObserver, observer } from "mobx-react-lite"
+import ReactDOM from "react-dom"
+
+const person = observable({
+    name: "John"
+})
+
+// named function is optional (for debugging purposes)
+const P1 = observer(function P1({ person }) {
+    return <h1>{person.name}</h1>
+})
+
+const P2 = ({ person }) => <Observer>{() => <h1>{person.name}</h1>}</Observer>
+
+const P3 = ({ person }) => {
+    return useObserver(() => <h1>{person.name}</h1>)
+}
+
+ReactDOM.render(
+    <div>
+        <P1 person={person} />
+        <P2 person={person} />
+        <P3 person={person} />
+    </div>
+)
+
+setTimeout(() => {
+    person.name = "Jane"
+}, 1000)
+```
+
+[![Edit MobX in React - observer pattern](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/4qrv105l47?fontsize=14)
+
+When using `observer`, `React.memo` will be applied automatically to the component, and `useObserver` will be applied under the hood..
+
+`useObserver` keeps the component tree flat as it doesn't introduce a higher order component, but to make props observable, the [`useAsObservableSource`](#useasobservablesourcetstate-t-t) hook should be used.
+
+`<Observer>` is slightly better optimized compared to `useObserver`, as it doesn't cause a double rendering when using `useAsObservableSource`, but does introduce an additional component in the component tree.
+
+### Introducing local observable state
+
+Local observable state can be introduced by using the [`useLocalStore`](#uselocalstoretinitializer---t-t) hook, that runs once to create an observable store. A quick example would be:
+
+```javascript
+import { useLocalStore, useObserver } from "mobx-react-lite"
+
+const P3 = () => {
+    const todo = useLocalStore(() => ({
+        title: "Test",
+        done: true,
+        toggle() {
+            this.done = !this.done
+        }
+    }))
+
+    return useObserver(() => (
+        <h1 onClick={todo.toggle}>
+            {todo.title} {todo.done ? "[DONE]" : "[TODO]"}
+        </h1>
+    ))
+})
+```
+
+When using `useLocalStore`, all properties of the returned object will be made observable automatically, getters will be turned into computed properties, and methods will be bound to the store and apply mobx transactions automatically.
+
+_Note: using `useLocalStore` is mostly beneficial for really complex local state, or to obtain more uniform code base. Note that using a local store might conflict with future React features like concurrent rendering._
+
+### More extensive example
```

Comment:
small problem here:

> `The following example combines all concepts mentioned so far [...] and useAsObservableProps to make the props observable`

because `useAsObservableProps` has not yet been mentioned in the readme. this is the first time `useAsObservableProps` appears